### ISSUE TITLE
'formality' is not supported for given 'target_lang'

### ISSUE THREAD

**Issue by gregzhadko** [DeepLcom/deepl-dotnet#31]
I encountered an error when attempting to use Ukrainian as the target language with the formality option.:

> Bad request, message: 'formality' is not supported for given 'target_lang'.

My code is

```
var options = new TextTranslateOptions { Formality = Formality.Less};
var textResults = await _translator.TranslateTextAsync(text, "en", "uk", options);
```

Is it possible for this to occur with other languages as well? Is there a way for me to anticipate this issue without having to manually check each language?

**Comment by daniel-jones-deepl** [comment]
Hi @gregzhadko, thanks for the question. Yes, only some target languages support the formality parameter. You can query which target languages support formality using [GetTargetLanguagesAsync()](https://github.com/DeepLcom/deepl-dotnet#listing-available-languages). Alternatively, you can specify the formality option as `Formality.PreferLess` or `Formality.PreferMore`; these options will fallback to the default formality if formality is unavailable for the target language. I hope that answers your question. 

I guess you read the project readme, is that right? Unfortunately our readme is a little unclear on this topic. We've made some improvements to our Python client readme, these changes haven't yet been applied to this project; we'll do that soon. In any case, you might find [this section](https://github.com/DeepLcom/deepl-python#text-translation-options) and [this section](https://github.com/DeepLcom/deepl-python#listing-available-languages) helpful. Although the parameter & function names differ in capitalisation, the functionality is the same.

**Comment by gregzhadko** [comment]
Hello @daniel-jones-deepl, 
Thank you for your response. I came across the [GetTargetLanguagesAsync()](https://github.com/DeepLcom/deepl-dotnet#listing-available-languages) method, but it seems inconvenient to use since I would have to retrieve the information every time I need to translate. Fortunately, your suggestion regarding `Formality.PreferLess` and `Formality.PreferMore` is precisely what I required. Initially, I was confused by their naming, as I assumed they represented another level of formality.  Perhaps documentation in the code or a better naming convention, such as `Formality.LessIfPossible`, could alleviate any further confusion. 
Thank you for your invaluable assistance.
